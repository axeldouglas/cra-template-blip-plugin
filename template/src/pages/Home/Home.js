import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import ConnectTo from '../../store/connect';

import logo from '../../assets/images/svg/blip-balloon.svg';
import Header from './components/Header';

const PAGE_TITLE = 'Plugins Template';
const PAGE_ICON = 'plugin';
const REPOSITORY_URL =
    'https://github.com/axeldouglas/cra-template-blip-plugin';

const Home = ({ dispatch }) => {
    const handleNavigation = useCallback((path, params = {}) => {
        window.open(REPOSITORY_URL, '_blank');
    }, []);

    return (
        <div className="ph1 ph4-m ph5-ns pb5">
            <Header
                title={PAGE_TITLE}
                icon={PAGE_ICON}
                onClick={() => handleNavigation()}
            />
            <div className="flex flex-column items-center justify-center bp-c-neutral-dark-city f5 h-100 mt5">
                <img src={logo} className="App-logo" alt="logo" />
                <p className="tc">
                    This is a basic plugin template.
                    <br />
                    Edit <code>src/pages/Home.js</code> and save to reload.
                </p>
            </div>
        </div>
    );
};

Home.propTypes = {
    dispatch: PropTypes.func
};

const mapStateToProps = (state, props) => ({
    ...props
});

export default ConnectTo(mapStateToProps)(Home);
