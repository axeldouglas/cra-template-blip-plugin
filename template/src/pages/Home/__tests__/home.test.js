import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Home from '../Home';

describe('Home page', () => {
    it('should renders default page', () => {
        render(<Home />);

        expect(
            screen.getByRole('heading', {
                name: /plugins template/i
            })
        ).toBeInTheDocument();
        expect(screen.getByText(/visit the repository/i)).toBeInTheDocument();
    });

    it('should matches with snapshot', () => {
        const { container } = render(<Home />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
