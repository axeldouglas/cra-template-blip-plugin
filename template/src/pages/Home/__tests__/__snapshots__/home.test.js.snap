// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should matches with snapshot 1`] = `
<div
  class="ph1 ph4-m ph5-ns pb5"
>
  <div
    class="flex flex-row items-center-ns justify-between w-100 pv3 mt2 bb bp-bc-neutral-medium-wave"
  >
    <div
      class="flex items-center"
    >
      <div
        data-testid="page-header-icon"
      >
        <bds-icon
          color="#505F79"
          name="plugin"
        />
      </div>
      <h2
        class="f3 ml2 mr1 bp-c-neutral-dark-city"
      >
        Plugins Template
      </h2>
    </div>
    <div
      class="flex items-center justify-end"
    >
      <div
        class="ml3"
      >
        <bds-button
          aria-label="bds-button"
          data-testid="my-plugin-button"
          icon="link"
          variant="ghost"
        >
          Visit the Repository
        </bds-button>
      </div>
    </div>
  </div>
  <div
    class="flex flex-column items-center justify-center bp-c-neutral-dark-city f5 h-100 mt5"
  >
    <img
      alt="logo"
      class="App-logo"
      src="blip-balloon.svg"
    />
    <p
      class="tc"
    >
      This is a basic plugin template.
      <br />
      Edit 
      <code>
        src/pages/Home.js
      </code>
       and save to reload.
    </p>
  </div>
</div>
`;
